### ISSUE TITLE
Use --exact flag when generating run configuration for a single test

### ISSUE THREAD

**Issue by matklad** [intellij-rust/intellij-rust#943]
https://github.com/rust-lang/rust/pull/38181

**Comment by maraisr** [comment]
@matklad  this issue still up for grabs, and if so, are you able to point me in the right direction?

**Comment by vlad20012** [comment]
The issue is still actual and up for grab. If you try to run a single test name `test` in this setup:
```rust
mod test {
    #[test]
    fn test(){
        println!("test");
    }
    #[test]
    fn test321(){
        println!("test321");
    }
}
```
both test are runned instead.

I think it's located somewhere [here](https://github.com/intellij-rust/intellij-rust/blob/7cd84c892bb48021b9d5157cdbb8dc54c269b026/src/main/kotlin/org/rust/cargo/runconfig/test/CargoTestRunConfigurationProducer.kt#L91).

**Comment by maraisr** [comment]
@vlad20012  do you think you can assign this job to me - happy to take it on!

Few questions though. What is the development workflow for this? Or is it pretty standard dev workflow for Intellij plugins?

**Comment by vlad20012** [comment]
I'm sorry, I could not understand the question. Could you please clarify it? However, perhaps you can find the answer in the [CONTRIBUTING.md](https://github.com/intellij-rust/intellij-rust/blob/master/CONTRIBUTING.md)

**Comment by vlad20012** [comment]
@maraisr And yes, you can grab the issue anytime :)

**Comment by maraisr** [comment]
Doesn't look like `--exact` is a flag anymore? 

Using the [above](https://github.com/intellij-rust/intellij-rust/issues/943#issuecomment-332696870) code block and running:

```bash
$ cargo test ::test321
    Finished dev [unoptimized + debuginfo] target(s) in 0.0 secs
     Running target/debug/deps/rust-9d6b4d08e5c8e0ac

running 1 test
test test::test321 ... ok

test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 1 filtered out
```

Only executes the one test anyway?

**Comment by vlad20012** [comment]
@maraisr try to run test named `test` instead of `test123`

**Comment by maraisr** [comment]
@vlad20012 I see! 

```bash
$ cargo test ::test
    Finished dev [unoptimized + debuginfo] target(s) in 0.0 secs
     Running target/debug/deps/rust-9d6b4d08e5c8e0ac

running 2 tests
test test::test ... ok
test test::test321 ... ok

test result: ok. 2 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out
```

but then why does this fail?
```bash
$ cargo test ::test --exact
error: Unknown flag: '--exact'

Usage:
    cargo test [options] [--] [<args>...]
```

My cargo version: **cargo 0.22.0-nightly**

**Comment by vlad20012** [comment]
@maraisr,
`test --package example_project --bin example_project test::test -- --exact` ?

**Comment by pasa** [comment]
Hi! I've created pull request #1868 for this issue. Please, have a look.

**Comment by pasa** [comment]
I guess this issue can be closed since the pull request #1868 has been merged.
